### PR TITLE
Py3: More changes

### PR DESCRIPTION
**Pull request by iarspider**
`process_pr` fix is just to remove error (one should always use `open` to open a file). There are more changes required to make `process_pr` fully py3-compatible: I see many calls to `.encode("ascii", "ignore")` and comparision of result with string. This will no longer work, since `encode` returns `bytes` object, which can't be compared to `str` (directly or with a regex). Do you remember why these calls were added?

### CODE REVIEW

**Comment by iarspider** [comment]
Ready for review

**Comment by smuzaffar** [comment]
>I see many calls to .encode("ascii", "ignore") and comparision of result with string. This will no longer work, since encode returns bytes object, which can't be compared to str (directly or with a regex). 

yes , github user name can contain non-ascii chars so one need encoding

**Comment by smuzaffar** [comment]
are all these formatting changes needed for py3? If not then please remove them , this will help with the review.

**Comment by iarspider** [comment]
@smuzaffar I have tested most of the files touched (except `es_ibs_log.py`, which requires log files, and `process_pr.py`, which requires a lot more changes), should be fine.

File: jenkins_callback.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def build_jobs(jenkins_url, jobs_data, headers = {}, user="cmssdt"):
              "Submit": "Build"
            }
     try:
-      data = urlencode(data)
+      data = urlencode(data).encode()
```
Comment (smuzaffar) [inline jenkins_callback.py:29]: @iarspider , does this work for both python2 and python3. This code is used by cmssdt-web too ( https://github.com/search?q=repo%3Acms-sw%2Fcmssdt-web%20jenkins_callback&type=code )

File: trigger_jenkins_job.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
-#!/usr/bin/env python
-from __future__ import print_function
+#!/usr/bin/env python3
```
Comment (smuzaffar) [inline trigger_jenkins_job.py:1]: were you able to trigger any jenkins job using this?

Comment (smuzaffar) [reply trigger_jenkins_job.py:1]: looks fine, I was able to trigger https://cmssdt.cern.ch/jenkins/view/Jenkins%20Tests/job/jenkins-test-job3/1/console

File: jenkins_callback.py (hunk 1 above)
Comment (smuzaffar) [reply jenkins_callback.py:29]: yes looks good

**Comment by smuzaffar** [comment]
@iarspider , for `es_ibs_logs.py` I am getting errors like
```
Working on  /data/sdt/SDT/jenkins-artifacts/HLT-Validation/CMSSW_13_3_X_2023-08-29-1100/el8_amd64_gcc11/OnLine_HLT_GRun_DATA.log
Traceback (most recent call last):
  File "./es_ibs_log.py", line 237, in <module>
    process_hlt_log(logFile)
  File "./es_ibs_log.py", line 118, in process_hlt_log
    for index, l in enumerate(f):
  File "/usr/lib64/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 3867: ordinal not in range(128)
```

**Comment by smuzaffar** [comment]
+externals
sample run of tests look good, lets get this in.

**Pull request merged**